src/core/simulator/entity/floor_entity.cpp

```diff
@@ -41,8 +41,8 @@ namespace argos {
       virtual CColor GetColorAtPoint(Real f_x,
                                      Real f_y) {
          /* Compute coordinates on the image */
-         UInt32 x = (f_x + m_cHalfArenaSize.GetX()) * m_fArenaToImageCoordinateXFactor;
-         UInt32 y = (f_y + m_cHalfArenaSize.GetY()) * m_fArenaToImageCoordinateYFactor;
+         UInt32 x = static_cast<UInt32>((f_x + m_cHalfArenaSize.GetX()) * m_fArenaToImageCoordinateXFactor);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a implicit float -> int conversion warning without the cast